File: abc_341/C/main.py
#!/usr/bin/env python3
H, W, N = map(int, input().split())
S = [None for _ in range(H)]
T = input()
for i in range(H):
    S[i] = input()

ans = 0
for i in range(H):
    for j in range(W):
        if S[i][j] == "#":
            continue
        x = i
        y = j
        ok = True
        for k in list(T):
            if k == "U":
                x -= 1
            elif k == "D":
                x += 1
            elif k == "L":
                y -= 1
            elif k == "R":
                y += 1
            if S[x][y] == "#":
                ok = False
                break
        if ok:
            # print(i, j, S[i][j])
            ans += 1
print(ans)

File: abc_341/C/main.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var tmp string
	var h, w, n int
	var s [][]string
	fmt.Scanf("%d %d %d", &h, &w, &n)
	fmt.Scanf("%s", &tmp)
	t := strings.Split(tmp, "")
	// fmt.Println(h, w, n)
	// fmt.Println(t)
	for i := 0; i < h; i++ {
		fmt.Scanf("%s", &tmp)
		s = append(s, strings.Split(tmp, ""))
	}
	// fmt.Printf("%+v\n", s)

	ans := 0
	for i := 0; i < h; i++ {
		for j := 0; j < w; j++ {
			if s[i][j] == "#" {
				continue
			}
			x := i
			y := j
			ok := true
			for _, c := range t {
				switch c {
				case "U":
					x--
				case "D":
					x++
				case "L":
					y--
				case "R":
					y++
				}
				if s[x][y] == "#" {
					ok = false
					break
				}
			}
			if ok {
				ans++
			}
		}
	}
	fmt.Println(ans)
}
